### ISSUE TITLE
Graphs - Play Totals not working

### ISSUE THREAD

**Issue by pax0707** [Tautulli/Tautulli#423]
Stuck at "loading chart".

2 different PMS Win 10 PCs.

**Comment by JonnyWong16** [comment]
Any errors or warnings in the logs?

**Comment by pax0707** [comment]
Nope.

```
2016-01-05 01:34:06 INFO    Scheduled background task: Refresh users list
2016-01-05 01:34:06 INFO    Scheduled background task: Check for active sessions
2016-01-05 01:34:06 INFO    Scheduled background task: Check for server response
2016-01-05 01:34:06 INFO    Scheduled background task: Check for recently added items
2016-01-05 01:34:06 INFO    Scheduled background task: Refresh Plex Server Name
2016-01-05 01:34:06 INFO    Scheduled background task: Refresh Plex Server URLs
2016-01-05 01:34:06 INFO    Scheduled background task: Check GitHub for updates
2016-01-05 01:34:05 INFO    Web server authentication is enabled, username is 'pax0707'
2016-01-05 01:34:05 INFO    Starting PlexPy web server on http://0.0.0.0:8181/
2016-01-05 01:34:05 INFO    Users list refreshed.
2016-01-05 01:34:03 INFO    Requesting users list refresh...
2016-01-05 01:34:03 INFO    Requesting name from server...
2016-01-05 01:34:03 INFO    Writing configuration to file
2016-01-05 01:34:03 INFO    Writing configuration to file
2016-01-05 01:34:03 INFO    Requesting URLs for server...
2016-01-05 01:34:03 INFO    PlexPy is up to date
2016-01-05 01:34:03 DEBUG   Latest version is 7e5dce1c14aae1b4829357897e46bc8718c3e124
2016-01-05 01:34:02 DEBUG   Requesting URL via GET method: https://api.github.com/repos/drzoidberg33/plexpy/commits/dev
2016-01-05 01:34:02 INFO    Retrieving latest version information from GitHub
2016-01-05 01:34:02 DEBUG   Git output: dev
2016-01-05 01:34:02 DEBUG   Trying to execute: "git rev-parse --abbrev-ref HEAD" with shell in D:\Plex\PlexPy
2016-01-05 01:34:02 DEBUG   Git output: 7e5dce1c14aae1b4829357897e46bc8718c3e124
2016-01-05 01:34:01 DEBUG   Trying to execute: "git rev-parse HEAD" with shell in D:\Plex\PlexPy
2016-01-05 01:34:01 INFO    Checking to see if the database has all tables....
```

**Comment by JonnyWong16** [comment]
Is this a new install?

**Comment by pax0707** [comment]
No.

It was working week or two ago.

**Comment by JonnyWong16** [comment]
Have you tried a force refresh (Ctrl+F5)?

**Comment by pax0707** [comment]
Yes.

I'm using Chrome 48.
Also tried in IE9, 10 and edge.

**Comment by JonnyWong16** [comment]
What do you get if you go to `http://localhost:8181/get_plays_per_month`?

**Comment by pax0707** [comment]
```
500 Internal Server Error

The server encountered an unexpected condition which prevented it from fulfilling the request.

Traceback (most recent call last):
  File "D:\Plex\PlexPy\lib/cherrypy\_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "D:\Plex\PlexPy\lib/cherrypy\lib\encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "D:\Plex\PlexPy\lib/cherrypy\_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "D:\Plex\PlexPy\plexpy\webserve.py", line 1028, in get_plays_per_month
    return json.dumps(result)
  File "C:\Python27\lib\json\__init__.py", line 243, in dumps
    return _default_encoder.encode(obj)
  File "C:\Python27\lib\json\encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "C:\Python27\lib\json\encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x9e in position 1: invalid start byte
```

**Comment by JonnyWong16** [comment]
Looks like there is something with a special character causing an error.

**Comment by pax0707** [comment]
Anything I can check?

**Comment by JonnyWong16** [comment]
I'm not sure. Do you mind sending me you `plexpy.db` file on the [forum](http://forums.plex.tv/profile/JonnyWong16) to take a look? Maybe something went wrong.

**Comment by pax0707** [comment]
Done.

**Comment by JonnyWong16** [comment]
Hmm, database is fine. Try resetting PlexPy by running `git reset --hard HEAD` from the command line in the PlexPy folder. You won't lose your settings or database.

**Comment by pax0707** [comment]
Tried, same error.

**Comment by JonnyWong16** [comment]
I'm out of ideas...

**Comment by Arcanemagus** [comment]
What do you get when you put the following code in [right here](https://github.com/drzoidberg33/plexpy/blob/v1.2.16/plexpy/webserve.py#L1018):

``` python
print vars(result)
```

**Comment by pax0707** [comment]
```
Traceback (most recent call last):
  File "lib/cherrypy\_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "lib/cherrypy\lib\encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "lib/cherrypy\_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "D:\Plex\PlexPy\plexpy\webserve.py", line 1018, in get_plays_per_month
    print vars(result)
TypeError: vars() argument must have __dict__ attribute
```

**Comment by Arcanemagus** [comment]
Try this instead:

``` python
import pprint
pprint.pprint(result)
```

**Comment by pax0707** [comment]
```
Traceback (most recent call last):
  File "lib/cherrypy\_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "lib/cherrypy\lib\encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "lib/cherrypy\_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "D:\Plex\PlexPy\plexpy\webserve.py", line 1022, in get_plays_per_month
    return json.dumps(result)
  File "C:\Python27\lib\json\__init__.py", line 243, in dumps
    return _default_encoder.encode(obj)
  File "C:\Python27\lib\json\encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "C:\Python27\lib\json\encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x9e in position 1: invalid start byte
```

**Comment by magloui** [comment]
I have same issue
in webserve.py line 1021
I add this  return json.dumps(result, encoding='latin1')

For me due to french setting 

Dec. 2015 = in french Déc.2015

![2016-01-07 23_04_47-plexpy - graphs](https://cloud.githubusercontent.com/assets/16600552/12184071/528ee3a6-b593-11e5-965f-6b6100e027a7.png)

**Comment by JonnyWong16** [comment]
Just as I expected. A special character.

**Comment by pax0707** [comment]
Yeah, that's it.
Croatia - iso8859_2 for me (Central and Eastern Europe)

```
{"series": [{"data": [0, 0, 0, 0, 0, 0, 0, 0, 0, 146, 209, 49], "name": "TV"}, {"data": [0, 0, 0, 0, 0, 0, 0, 0, 0, 72, 70, 12], "name": "Movies"}, {"data": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "name": "Music"}], "categories": ["vlj 2015", "o\u009eu 2015", "tra 2015", "svi 2015", "lip 2015", "srp 2015", "kol 2015", "ruj 2015", "lis 2015", "stu 2015", "pro 2015", "sij 2016"]}
```

Play by period and stream info show English dates, play totals Croatian but it works.

**Comment by JonnyWong16** [comment]
Does this fix the problem? Revert the other changes.

Change this line: https://github.com/drzoidberg33/plexpy/blob/master/plexpy/graphs.py#L314
to

```
categories.append(dt.strftime('%b %Y').decode(locale.getlocale()[1]))
```

**Comment by magloui** [comment]
after change
  File "C:\plexpy\plexpy\graphs.py", line 314, in get_total_plays_per_month
    categories.append(dt.strftime('%b %Y')).decode(locale.getlocale()[1])
AttributeError: 'NoneType' object has no attribute 'decode'

**Comment by JonnyWong16** [comment]
The `.decode` part is inside the bracket.

Might also need `import locale` on line 19.

**Comment by magloui** [comment]
I correct my mistake, and now

  File "C:\plexpy\plexpy\graphs.py", line 314, in get_total_plays_per_month
    categories.append(dt.strftime('%b %Y').decode(locale.getlocale()[1]))
NameError: global name 'locale' is not defined

**Comment by JonnyWong16** [comment]
Did you also add `import locale` on line 19?

**Comment by pax0707** [comment]
Graph now works.
Still showing localized date.

**Comment by magloui** [comment]
after add the `import locale`, is work like a charm
Many thanks

**Comment by JonnyWong16** [comment]
Great, now I know how to fix them when I get home.

You guys will need to undo all the changes before the next release or you won't be able to update PlexPy. The easiest way would be to run `git reset --hard HEAD` from the PlexPy folder in the command line.

**Comment by JonnyWong16** [comment]
Fixed in v1.3.0.
